client/main.lua

```diff
@@ -75,3 +105,17 @@ lib.callback.register('qbx_fireworks:client:useFirework', function(asset)
         return false
     end
 end)
+
+RegisterNetEvent('qbx_fireworks:client:startShow', function(show)
+    if show ~= nil then
+        for i=1, #sharedConfig.shows[show] do -- Sequences
+            for j=1, #sharedConfig.shows[show][i].fireworks do -- Individual
+                TriggerServerEvent('qbx_fireworks:server:spawnShowObject', sharedConfig.shows[show][i].fireworks[j].asset, sharedConfig.shows[show][i].fireworks[j].coords)
+                startFireworkShow(sharedConfig.shows[show][i].fireworks[j].asset, sharedConfig.shows[show][i].fireworks[j].coords, sharedConfig.shows[show][i].fireworks[j].height)
+                Wait(sharedConfig.shows[show][i].fireworks[j].wait)
+            end
+            Wait(sharedConfig.shows[show][i].masterWait)
+        end
+    end
```

Comment:
using variables will make the code more readable. Avoid referencing the same table value twice.
https://manason.github.io/effective-fivem-lua/tables/#extract-duplicate-table-dereferences-into-local-variables

Reply:
outdated